pex_reactor.go

```diff
@@ -195,26 +255,40 @@ func (pexR *PEXReactor) ensurePeers() {
 		if picked == nil {
 			continue
 		}
-		toDial.Set(picked.IP.String(), picked)
+		toDial[picked.IP.String()] = picked
 	}
 
 	// Dial picked addresses
-	for _, item := range toDial.Values() {
+	for _, item := range toDial {
 		go func(picked *NetAddress) {
-			_, err := pexR.Switch.DialPeerWithAddress(picked, false)
+			_, err := r.Switch.DialPeerWithAddress(picked, false)
 			if err != nil {
-				pexR.book.MarkAttempt(picked)
+				r.book.MarkAttempt(picked)
 			}
-		}(item.(*NetAddress))
+		}(item)
 	}
 
 	// If we need more addresses, pick a random peer and ask for more.
-	if pexR.book.NeedMoreAddrs() {
-		if peers := pexR.Switch.Peers().List(); len(peers) > 0 {
+	if r.book.NeedMoreAddrs() {
+		if peers := r.Switch.Peers().List(); len(peers) > 0 {
 			i := rand.Int() % len(peers)
 			peer := peers[i]
 			log.Info("No addresses to dial. Sending pexRequest to random peer", "peer", peer)
-			pexR.RequestPEX(peer)
+			r.RequestPEX(peer)
+		}
+	}
+}
+
+func (r *PEXReactor) flushMsgCountByPeer() {
+	ticker := time.NewTicker(msgCountByPeerFlushInterval)
+
+	for {
+		select {
+		case <-ticker.C:
+			r.msgCountByPeer = make(map[string]uint16)
```

Comment:
does it have to be? peer has only 1 MConn ; MConn calls `onReceive` in the same thread (blocking)